node/node.go

```diff
@@ -873,6 +873,17 @@ func NewNode(config *cfg.Config,
 		return nil, fmt.Errorf("could not create addrbook: %w", err)
 	}
 
+	for _, addr := range splitAndTrimEmpty(config.P2P.BootstrapPeers, ",", " ") {
```

Comment:
As this is hard to unit-test (code in Node usually is), would it make sense to add support for this in the e2e manifest (no need to touch the generator) so we can test this in, e.g., `networks/ci.toml`?

Reply:
I didn't get the point here.

Reply:
The addresses here are directly passed to the address book, the address book unit tests should cover this. If not, the PEX reactor unit test, which requires setting up the address book, should also cover this.

Reply:
> I didn't get the point here.

As a general rule, when we add production code, we need to add some (unit) test code that hits the new code.
Here, I think unit testing would be cumbersome and pointless (but I might be wrong, as I didn't look around into UTs that test neighboring code... it may be simpler), so I was suggesting to add the config option as part of `networks/ci.toml` so that it's part of e2e

> The addresses here are directly passed to the address book, the address book unit tests should cover this. If not, the PEX reactor unit test, which requires setting up the address book, should also cover this.

Can you point at the test cases that are testing this new code?

Reply:
There is no test unit covering this flag specifically. In particular because this is not a flag that is considered by the p2p layer.

What I meant is that there are test cases covering the `addrBook.AddAddress()` method, which is a very basic method of the PEX implementation, that is invoked for all addresses parsed from this config flag.

Reply:
> so I was suggesting to add the config option as part of networks/ci.toml so that it's part of e2e

Ok, I will take a look on this file.

Reply:
Ok, found it, but this is a specific config triggering some scenarios. I don't see the use of this flag in the network setup/topology enforced in those runs. Actually, those runs do not cover the user case of this new flag, as node rely on persistent peers and a seed node for establishing connections. The option there would be to replace a persistent peer in a given node by a bootstrap peer, but this would mean changing the semantics of the setup, as the node will not necessarily connect to the configured bootstrap peer, as it can retrieve other addresses from the nodes configured as persistent or seeds beforehand.

Reply:
OK, if you think it makes no sense to test this using e2e, then we're back with unit tests.

I took a closer look at relevant existing UTs. There is `TestNodeNewNodeCustomReactors` which is testing the `NewNode` function where your new code is. So I see several options:

* You can create a similar test case, next to it, where you modify the config before passing it to `NewNode` so that in contains relevant bootstrap peer config. The easiest is to feed incorrect config, to exercise the error paths introduced by the new code (where you are returning an error).
* For testing the happy path, I think it's best to modify `TestP2PConfig` to include correct bootstrap config, and then re-running all UTs and make sure they all pass. Alternatively, you can just test the happy path as part of the new UT where you're testing the error paths